src/main/webapp/app/shared/user-settings/user-settings.route.ts

```diff
@@ -0,0 +1,27 @@
+import { Routes } from '@angular/router';
+import { UserSettingsContainerComponent } from 'app/shared/user-settings/user-settings-container/user-settings-container.component';
+import { AccountInformationComponent } from 'app/shared/user-settings/account-information/account-information.component';
+import { NotificationSettingsComponent } from 'app/shared/user-settings/notification-settings/notification-settings.component';
+
+export const userSettingsState: Routes = [
+    {
+        path: 'user-settings',
+        component: UserSettingsContainerComponent,
+        children: [
+            {
+                // redirects to account information when user-settings are opened
+                path: '',
+                pathMatch: 'full',
+                redirectTo: 'account',
+            },
+            {
+                path: 'account',
+                component: AccountInformationComponent,
+            },
+            {
+                path: 'notifications',
+                component: NotificationSettingsComponent,
+            },
+        ],
+    },
```

Comment:
I think this is missing route access logic. This might be why trying to access link **/user-settings/notifications** while not being logged in it will not redirect to root route. This file might be helpful: [here](https://github.com/ls1intum/Artemis/blob/bbdfa6c60a010bc276b09909d9a04fe55bc028ef/src/main/webapp/app/exercises/text/manage/tutor-effort/tutor-effort-statistics.route.ts)

Reply:
Great, thank you very much for your help I managed to fix this. 👍